### DISCUSSION TITLE
Future of this integration

### DISCUSSION THREAD

**Discussion by petretiandrea** [petretiandrea/home-assistant-tapo-p100#740] [Polls]
There is a version on HA based on the creators of the kasa library, I wanted to open a poll on whether to continue the work of this library. My idea was to make this ingradation the official HA one, but I would like to understand from users if you actually notice differences or have preferences

**Comment by yoshimo** [comment]
I find it a waste of time when multiple people work on their own project instead of working together on a common one.
So while i didn't notice differences for myself, i think having just one common integration in upstream Home-Assistant would make it more efficient.

**Reply by seba1337** [reply]
The official one supports less devices AFAIK.

**Reply by mcatkins** [reply]
I could only get the smart plug to work, but nothing on a h100 hub

**Comment by jamesonuk** [comment]
I did try the official one and it worked OK for the lights and plugs I have, but had less features.   
lights did not support effects and plugs were missing some sensors (although I think they were signal strength ones so not the end of the world).   I do suspect all the hub stuff doesn't work with the core integration

I don't know what the code bases are like in terms of quality and debt but they both seem reasonable from a cursory glance.

I am not fussed as to which one is used but having these features in core integration would be great.

**Reply by yoshimo** [reply]
I also feel that some parts of the official integrations are off, like different sensors showing the same value, but this might be true the other way round as well. In the end, we should converge to one code base and apply fixes there instead of duplicating work

**Comment by MakaanPL** [comment]
I think it would be great to integrate (pun intended) both in HA, if not conflicting, and decide on using single codebase going forward.

At least for KE100 thermostat valves, some features are present in one but not in the other, and vice versa. For example they have signal level and cloud connection status, but no child lock and maybe frost protection, also implemented "heat" (idle + probably other) statuses while yours can be either "heat" or "off" if actually heating or not (more straightforward, but not as robust).
